File: frontend/src/api/stats.ts
import { apiClient } from '@/lib/apiClient';

export interface PeriodStats {
  workouts: number;
  volume_kg: number;
  records: number;
}

export interface StrengthGain {
  exercise_id: string;
  exercise_name: string;
  first_best_kg: number;
  current_best_kg: number;
  gain_kg: number;
  gain_pct: number;
}

export interface StatsOverview {
  first_workout_at: string | null;
  total_workouts: number;
  total_sets: number;
  total_reps: number;
  total_volume_kg: number;
  this_month: PeriodStats;
  last_month: PeriodStats;
  strength_gains: StrengthGain[];
  next_milestone: { target: number; remaining: number; previous: number } | null;
}

/** Months are computed in the device's timezone so "this month" matches the user's calendar. */
export async function fetchStatsOverview(): Promise<StatsOverview> {
  const tz = Intl.DateTimeFormat().resolvedOptions().timeZone || 'UTC';
  return apiClient.get<StatsOverview>(`/stats/overview?tz=${encodeURIComponent(tz)}`);
}

File: backend/app/schemas/stats.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class PeriodStats(BaseModel):
    workouts: int
    volume_kg: float
    records: int


class StrengthGain(BaseModel):
    exercise_id: UUID
    exercise_name: str
    first_best_kg: float
    current_best_kg: float
    gain_kg: float
    gain_pct: float


class Milestone(BaseModel):
    target: int
    remaining: int
    previous: int


class StatsOverview(BaseModel):
    first_workout_at: datetime | None
    total_workouts: int
    total_sets: int
    total_reps: int
    total_volume_kg: float
    this_month: PeriodStats
    last_month: PeriodStats
    strength_gains: list[StrengthGain]
    next_milestone: Milestone | None
